Header header
uint8[] data
uint32 size